extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetSchemaReader.java

```diff
@@ -166,10 +167,23 @@ public static ParquetInstructions readParquetSchema(
                 return column.getPath();
             });
             final String colName;
-            final String mappedName = readInstructions.getColumnNameFromParquetColumnName(parquetColumnName);
-            if (mappedName != null) {
-                colName = mappedName;
-            } else {
+            COL_NAME: {
+                if (fieldId != null) {
+                    final List<String> columnNames =
+                            readInstructions.getColumnNamesFromParquetFieldId(fieldId.intValue());
+                    if (columnNames.size() == 1) {
+                        colName = columnNames.get(0);
+                        break COL_NAME;
+                    } else if (columnNames.size() > 1) {
+                        // TODO: how should we handle this? Ignore?
+                        // throw new IllegalArgumentException();
```

Comment:
I feel that this should be an error in ParquetInstructions, right when the user sets it instead of here

Reply:
This whole code path is very smelly; I want to go through a larger refactoring that would alleviate the need to make these types of calls in the first place. This code path is only hit when inferring the TableDefinition, so I don't think it should be an error to set the same field id multiple times in general. We _have_ set it up this way with parquet column names, but we shouldn't technically need to do that either - every little modelling mismatch we present is a small papercut that can lead to larger modelling problems at higher layers IMO.

I would be _okay_ throwing an error here or silently ignoring wrt inferrence. Ideally, the user would be able to choose the behavior they desire. The structure of ParquetInstructions / builder makes that tedious (I wish we could redo it w/ Immutables and saner structures).

I'll change this to throw an error here, with a note we could think about exposing option to silently ignore if that's what the user wants.